utils/python/CIME/system_test.py

```diff
@@ -606,12 +608,12 @@ def _consumer(self, test, test_phase, phase_method):
                      (test_phase, test, elapsed_time, status)
         if not success:
             status_str += "    Case dir: %s\n" % self._get_test_dir(test)
-        sys.stdout.write(status_str)
+        logger.info(status_str)
 
         # On batch systems, we want to immediately submit to the queue, because
         # it's very cheap to submit and will get us a better spot in line
         if (success and not self._no_run and not self._no_batch and test_phase == MODEL_BUILD_PHASE):
-            sys.stdout.write("Starting %s for test %s with %d procs\n" % (RUN_PHASE, test, 1))
```

Comment:
logger is thread safe and in fact at debug level will tell you which thread printed each message.